flycheck-hledger.el

```diff
@@ -77,39 +77,39 @@ https://hledger.org/hledger.html#check."
 
    ;; hledger 1.26 assertions (:LINE:COL)
    (error
-    bol "hledger: Error: balance assertion: " (file-name (minimal-match (one-or-more (not ":")))) ":" line ":" column "\n"
-    (message (one-or-more bol (zero-or-more nonl) "\n")))
+    bol "hledger" (optional ".exe") ": Error: balance assertion: " (file-name) ":" line ":" column line-end
+    (message (one-or-more "\n" (zero-or-more nonl) line-end)))
```

Comment:
I'm surprised that this works on Windows because the end of line is supposed to consists of 2 characters `\r\n` but only `\n` is read here

Reply:
I’m not sure why either. The behaviour here around newlines doesn’t make a lot of sense to me.

Reply:
from what I understand `line-end` matches the empty string at the end of the line. This means that there is still the end of line character to match after `line-end` as matched. This doesn't explain why the code doesn't need to match `\r` though... If it works for you I guess it's good enough.